rest/route/agent_test.go

```diff
@@ -84,3 +96,151 @@ func TestAgentCedarConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestAgentSetup(t *testing.T) {
+	for tName, tCase := range map[string]func(ctx context.Context, t *testing.T, rh *agentSetup, s *evergreen.Settings){
+		"FactorySucceeds": func(ctx context.Context, t *testing.T, rh *agentSetup, s *evergreen.Settings) {
+			copied := rh.Factory()
+			assert.NotZero(t, copied)
+			_, ok := copied.(*agentSetup)
+			assert.True(t, ok)
+		},
+		"ParseSucceeds": func(ctx context.Context, t *testing.T, rh *agentSetup, s *evergreen.Settings) {
+			req, err := http.NewRequest(http.MethodGet, "https://example.com/rest/v2/agent/setup", nil)
+			require.NoError(t, err)
+			assert.NoError(t, rh.Parse(ctx, req))
+		},
+		"RunSucceeds": func(ctx context.Context, t *testing.T, rh *agentSetup, s *evergreen.Settings) {
+			resp := rh.Run(ctx)
+			require.NotZero(t, resp)
+			assert.Equal(t, http.StatusOK, resp.Status())
+
+			data, ok := resp.Data().(apimodels.AgentSetupData)
+			require.True(t, ok)
+			assert.Equal(t, data.SplunkServerURL, s.Splunk.ServerURL)
+			assert.Equal(t, data.SplunkClientToken, s.Splunk.Token)
+			assert.Equal(t, data.SplunkChannel, s.Splunk.Channel)
+			assert.Equal(t, data.S3Bucket, s.Providers.AWS.S3.Bucket)
+			assert.Equal(t, data.S3Key, s.Providers.AWS.S3.Key)
+			assert.Equal(t, data.S3Secret, s.Providers.AWS.S3.Secret)
+			assert.Equal(t, data.LogkeeperURL, s.LoggerConfig.LogkeeperURL)
+		},
+		"ReturnsEmpty": func(ctx context.Context, t *testing.T, rh *agentSetup, s *evergreen.Settings) {
+			*s = evergreen.Settings{}
+			resp := rh.Run(ctx)
+			require.NotZero(t, resp)
+			assert.Equal(t, http.StatusOK, resp.Status())
+
+			data, ok := resp.Data().(apimodels.AgentSetupData)
+			require.True(t, ok)
+			assert.Zero(t, data)
+		},
+	} {
+		t.Run(tName, func(t *testing.T) {
+			ctx, cancel := context.WithCancel(context.Background())
+			defer cancel()
+
+			s := &evergreen.Settings{
+				Splunk: send.SplunkConnectionInfo{
+					ServerURL: "server_url",
+					Token:     "token",
+					Channel:   "channel",
+				},
+				Providers: evergreen.CloudProviders{
+					AWS: evergreen.AWSConfig{
+						S3: evergreen.S3Credentials{
+							Bucket: "bucket",
+							Key:    "key",
+							Secret: "secret",
+						},
+						TaskSync: evergreen.S3Credentials{
+							Bucket: "bucket",
+							Key:    "key",
+							Secret: "secret",
+						},
+					},
+				},
+				LoggerConfig: evergreen.LoggerConfig{
+					LogkeeperURL: "logkeeper_url",
+				},
+			}
+
+			r, ok := makeAgentSetup(s).(*agentSetup)
+			require.True(t, ok)
+
+			tCase(ctx, t, r, s)
+		})
+	}
+}
+
+func TestDownstreamParams(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if err := db.ClearCollections(patch.Collection, task.Collection, host.Collection); err != nil {
```

Comment:
nit: this could be simplified down to `require.NoError(db.ClearCollections(...))`.